ingest/ingest-datasets.py

```diff
@@ -0,0 +1,154 @@
+# In development of utilizing sepatate .py files for common functions
+
+
+#########################################################
+# from SPARQLWrapper import SPARQLWrapper, JSON
+# from rdflib import Namespace, RDF
+# import json
+# import requests
+# import multiprocessing
+# import itertools
+# # from itertools import chain
+# import functools
+# import argparse
+# import warnings
+# import pprint
+#########################################################
+from ingestHelpers import *
+#########################################################
+
+# Global variables for the ingest process for: ***dataset***
+get_datasets_query = load_file("queries/listDatasets.rq")
+describe_dataset_query = load_file("queries/describeDataset.rq")
+_index = "dco"
+_type = "dataset"
+
+
+
+# describe_dataset: used by create_dataset_doc
+def describe_dataset(endpoint, dataset):
+    q = describe_dataset_query.replace("?dataset", "<" + dataset + ">")
+    return describe(endpoint, q)
+
+
+
+# create_dataset_doc: used by process_dataset
+def create_dataset_doc(dataset, endpoint):
+    graph = describe_dataset(endpoint=endpoint, dataset=dataset)
+    # graph = describe_object_function(endpoint=endpoint, dataset=dataset)
+
+    ds = graph.resource(dataset)
+
+    try:
+        title = ds.label().toPython()
+    except AttributeError:
+        print("missing title:", dataset)
+        return {}
+
+    dco_id = list(ds.objects(DCO.hasDcoId))
+    dco_id = str(dco_id[0].identifier) if dco_id else None
+
+    doc = {"uri": dataset, "title": title, "dcoId": dco_id}
+
+    doi = list(ds.objects(BIBO.doi))
+    doi = doi[0].toPython() if doi else None
+    if doi:
+        doc.update({"doi": doi})
+
+    abstract = list(ds.objects(BIBO.abstract))
+    abstract = abstract[0].toPython() if abstract else None
+    if abstract:
+        doc.update({"abstract": abstract})
+
+    most_specific_type = list(ds.objects(VITRO.mostSpecificType))
+    most_specific_type = most_specific_type[0].label().toPython() \
+        if most_specific_type and most_specific_type[0].label() \
+        else None
+    if most_specific_type:
+        doc.update({"mostSpecificType": most_specific_type})
+
+    publication_year = list(ds.objects(DCO.yearOfPublication))
+    publication_year = publication_year[0] if publication_year else None
+    if publication_year:
+        doc.update({"publicationYear": str(publication_year)})
+
+    dco_communities = get_dco_communities(ds)
+    if dco_communities:
+        doc.update({"dcoCommunities": dco_communities})
+
+    # portal_groups
+    portal_groups = get_portal_groups(ds)
+    if portal_groups:
+        doc.update({"portalGroups": portal_groups})
+
+    # projects NOT WORKING YET
+    projects = get_projects_of_dataset(ds)
+    if projects:
+        doc.update({"projects": projects})
+
+    # dataType
+    data_types = get_data_types(ds)
+    if data_types:
+        doc.update({"dataTypes": data_types})
+
+    # cites NOT WORKING YET
```

Comment:
These are just out-dated comments in development. I will re-do all the comments!